numbat/src/help.rs

```diff
@@ -0,0 +1,146 @@
+/**
+ * Print a help, linking the documentation and live-running some examples
+ * in an isolated context
+ */
+use crate::markup as m;
+use crate::module_importer::BuiltinModuleImporter;
+use crate::pretty_print::PrettyPrint;
+use crate::resolver::CodeSource;
+use crate::{Context, InterpreterResult, NumbatError};
+use crate::{InterpreterSettings, NameResolutionError, Type};
+
+use std::sync::{Arc, Mutex};
+
+fn evaluate_example(context: &mut Context, input: &str, pretty_print: bool) -> m::Markup {
+    let statement_output: Arc<Mutex<Vec<m::Markup>>> = Arc::new(Mutex::new(vec![]));
+    let statement_output_c = statement_output.clone();
+    let mut settings = InterpreterSettings {
+        print_fn: Box::new(move |s: &m::Markup| {
+            statement_output_c.lock().unwrap().push(s.clone());
+        }),
+    };
+
+    let (result, registry) = {
+        let registry = context.dimension_registry().clone(); // TODO: get rid of this clone
+        (
+            context.interpret_with_settings(&mut settings, input, CodeSource::Internal),
+            registry,
+        )
+    };
+
+    let mut full_output = m::empty();
+
+    match result {
+        Ok((statements, interpreter_result)) => {
+            if pretty_print {
+                full_output += statements
+                    .iter()
+                    .fold(m::empty(), |accumulated_mk, statement| {
+                        accumulated_mk
+                            + m::nl()
+                            + m::whitespace("  ")
+                            + statement.pretty_print()
+                            + m::nl()
+                    });
+            }
+
+            match interpreter_result {
```

Comment:
Hm. This code is now repeated (more or less) three times: in numbat-cli/src/main.rs, in numbat-wasm/src/lib.rs, and now here. It would be great if we could move this to a common place inside the `numbat` crate.

Reply:
I think I got this centralization working[^1], please look to see if it makes sense and is "clean".

[^1]: With the caveat that I still haven't tried running the wasm yet.